tools/c7n_azure/tests/azure_common.py

```diff
@@ -105,22 +130,103 @@ def request_callback(self, request):
             return None
         return request
 
+    def _response_callback(self, response):
+        if not hasattr(self, 'cassette') or os.path.isfile(self.cassette._path):
```

Comment:
Why do you need this? I think better to leave some comments here

Reply:
Agreed I am refactoring that now so its more clear and usable.  Basically tells you if you are recording or playback right now.

Reply:
its worth looking at how it was done here `self.recording = True | False` https://github.com/cloud-custodian/cloud-custodian/pull/3881/files

Reply:
`if os.environ.get(constants.ENV_ACCESS_TOKEN) == "fake_token":` isn't really a sufficient indicator we aren't recording though, its just a flag I set for CI but we also aren't recording when you have valid authentication and a cassette already exists which ends up being the most common scenario during development.  

I refactored a bit but its still tricky as you can't even do this in setup properly because the base class setup is both responsible for finding out if we are recording *and* it calls our callbacks which need to know.  You could have the callback set a flag the first time it saw a response if you really want to reduce filesystem checks.